DeskCheck/DeskCheck/Controllers/DesksController.cs

```diff
@@ -13,29 +13,47 @@ namespace DeskCheck.Controllers
     public class DesksController : ControllerBase
     {
         private readonly DeskCheckContext _context;
-        private IEnumerable<Desk> desks;
+        private List<Desk> desks;
 
         public DesksController(DeskCheckContext context)
         {
             _context = context;
+            desks = new List<Desk>();
+            string fpath = @".\DeskJsons\desks.json";
+            using (StreamReader sr = new StreamReader(fpath))
+            {
+                string json = sr.ReadToEnd();
+                desks = JsonConvert.DeserializeObject<List<Desk>>(json);
+            }
+
+            Random rng = new Random();
+            foreach(Desk d in desks)
+            {
+                d.CO2 += rng.Next(-100, 100);
+                d.temp += rng.Next(-5, 5);
+            }
+        }
+
+        public Desk NewDesk(int dnum)
+        {
             var rng = new Random();
-            desks = Enumerable.Range(1, 5).Select(index => new Desk
+            Desk d = new Desk
             {
-                deskID = index,
-                temperature = rng.Next(10, 30),
+                deskID = dnum,
+                temp = rng.Next(10, 30),
                 CO2 = rng.Next(200, 500), // Parts Per Million (PPM)
                 floor = rng.Next(0, 5),
                 X = rng.Next(50, 90),
                 Y = rng.Next(50, 90),
-                registered = rng.Next(0, 2) == 1
-            })
-            .ToArray();
+            };
+            return d;
         }
 
         // GET: desk/getAll
         [HttpGet("getAll")]
-        public IEnumerable<Desk> GetDesks()
+        public List<Desk> GetDesks()
         {
+            Console.WriteLine(desks);

```

Comment:
Might be a good idea to remove the Console.WriteLine. Things like this can cause large logs when deployed